lib/ffi/sdr_funcs.go

```diff
@@ -780,6 +806,11 @@ func (sb *SealCalls) SyntheticProofs(ctx context.Context, task *harmonytask.Task
 		}
 	}
 
+	// Generate unsealed copy before clearing cache
+	if err = sb.GenerateUnsealedSector(ctx, sector, fspaths, pathIDs, keepUnsealed); err != nil {
+		return xerrors.Errorf("generating unsealed sector: %w", err)
+	}
+
 	if err = ffi.ClearCache(uint64(ssize), fspaths.Cache); err != nil {
```

Comment:
Are you sure ClearCache will work when tree-d is not present? I imagine it should have no issues, but I'm actually not 100% certain, so we should test this first.